flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -223,17 +228,42 @@ public void run() {
 							}
 						}
 					}
-					if (updateOverview) {
-						updateJobOverview(webOverviewDir, webDir);
-						for (int x = 0; x < numFetchedArchives; x++) {
-							numArchivedJobs.countDown();
-						}
-					}
 				}
+
+				if (!jobsToRemove.isEmpty()) {
+					operations.addAll(removeDeletedJobs(jobsToRemove));
```

Comment:
maybe you can get rid of the operations list and fire events right away?

Reply:
The original version implemented this approach but it fails test. There was race condition between producing of event and refreshing content of jobs overview.

In this approach are events produced after update of jobs overview thus tests pass.